pkg/cmd/pulumi/deployment_settings_config.go

```diff
@@ -807,41 +838,34 @@ func configureOidcAzure(sd *workspace.ProjectStackDeployment) error {
 	var err error
 
 	if sd.DeploymentSettings.Operation.OIDC.Azure.ClientID != "" {
-		clientID, err = cmdutil.ReadConsoleWithDefault("Azure client id", sd.DeploymentSettings.Operation.OIDC.Azure.ClientID)
-	} else {
-		clientID, err = cmdutil.ReadConsole("Azure client id")
+		clientID = sd.DeploymentSettings.Operation.OIDC.Azure.ClientID
 	}
+
+	clientID, err = promptForValue(d.Yes, "Azure client id", clientID, false,
+		ValidateGenericInputNonEmpty, *d.DisplayOptions)
 	if err != nil {
 		return err
 	}
-	if clientID == "" {
-		return errors.New("Client id is required")
-	}
 
 	if sd.DeploymentSettings.Operation.OIDC.Azure.TenantID != "" {
-		tenantID, err = cmdutil.ReadConsoleWithDefault("Azure tenant id", sd.DeploymentSettings.Operation.OIDC.Azure.TenantID)
-	} else {
-		tenantID, err = cmdutil.ReadConsole("Azure tenant id")
+		tenantID = sd.DeploymentSettings.Operation.OIDC.Azure.TenantID
 	}
+
+	tenantID, err = promptForValue(d.Yes, "Azure tenant id", tenantID, false,
+		ValidateGenericInputNonEmpty, *d.DisplayOptions)
 	if err != nil {
 		return err
 	}
-	if tenantID == "" {
-		return errors.New("Tenant id is required")
-	}
 
 	if sd.DeploymentSettings.Operation.OIDC.Azure.SubscriptionID != "" {
-		subscriptionID, err = cmdutil.ReadConsoleWithDefault("Azure subscription id",
-			sd.DeploymentSettings.Operation.OIDC.Azure.SubscriptionID)
-	} else {
-		subscriptionID, err = cmdutil.ReadConsole("Azure subscription id")
+		subscriptionID = sd.DeploymentSettings.Operation.OIDC.Azure.SubscriptionID
 	}
+
+	subscriptionID, err = promptForValue(d.Yes, "Azure subscription id", subscriptionID, false,
```

Comment:
```suggestion
	subscriptionID, err = promptForValue(d.Yes, "Azure subscription ID", subscriptionID, false,
```
same elsewhere -- we should use `ID` in user-facing text IMO

Reply:
done